### PR TITLE
fix: upper case root sequence for ancestral reconstruction

### PR DESCRIPTION
**Pull request by rneher**
I ran into this issue when the root sequence was provided as lower case. All lower-to-upper case changes were attached to the root. This should fix it, but there might be better solutions.

### CODE REVIEW

**Comment by corneliusroemer** [comment]
Thanks! Seems like a strictly beneficial change, better solutions can be added later.

CI failing is because of monkeypox restructure, I will take mpox out of augur CI

**Comment by corneliusroemer** [comment]
Merging this then CI should work again: https://github.com/nextstrain/augur/pull/1324

**Comment by corneliusroemer** [comment]
Added a test that failed before the fix so that alternative fixes are easy to test and we don't regress in the future.
https://github.com/nextstrain/augur/pull/1323/commits/5d8bc27669c962bfcc2e1921f870cdf02d6c09a6

File: augur/ancestral.py
Comment (victorlin) [inline augur/ancestral.py:1]: I don't have enough background to understand why the change is needed. Is it because, for the purpose of ancestral reconstruction, "[soft-masking](https://bioinformatics.stackexchange.com/a/231)" in the root sequence is irrelevant and should be ignored?

File: CHANGES.md
Comment (victorlin) [inline CHANGES.md:1]: The current commits on this branch are hard for me to follow. Suggested alternative, by rebasing onto latest `master` and switching the order around: https://github.com/nextstrain/augur/compare/5aba7ec9b3b1c3c676c3f2b850d1148061de7340...07fa7b5a1a86414d066a51c0b57208b5ac3684f7

**Review by victorlin — APPROVED** [review]

File: CHANGES.md
Comment (corneliusroemer) [reply CHANGES.md:1]: Just look at the diff vs master - I don't think commit order matters here, it's just a few line changes in general anyways

File: augur/ancestral.py
Comment (corneliusroemer) [reply augur/ancestral.py:1]: I didn't know about soft vs hard masking. Indeed, we don't ever treat upper and lower case nucleotides differently anywhere in augur - at least not as far as I'm aware.

File: CHANGES.md
Comment (victorlin) [reply CHANGES.md:1]: Yeah, this is nitpicky and non-blocking! Just want to elaborate on my thoughts here.

My opinion is that "diff vs master" is only applicable if doing a squash merge, where the commits from the PR branch are irrelevant. Without the squash merge, there's unnecessarily verbose commit history which describes the ad-hoc iterative process while working on the PR, rather than the changes that the PR is actually trying to make.

To draw it out, merging as-is will lead to history that looks like this:

```
*   3c1b5e38 (HEAD -> master) Merge branch 'fix/root-upper-case'
|\  
| * f65b47c7 (origin/fix/root-upper-case, fix/root-upper-case) Add changelog entry
| *   7d97e078 Merge branch 'failing-test-for-1323' into fix/root-upper-case
| |\  
| | * 5d8bc276 Add failing test for #1323
| * | 1c78c885 Merge branch 'disable-monkeypox-ci' into fix/root-upper-case
| |\| 
| | * d5e5dc32 Disable monkeypox CI until pathogen-repo-ci supports custom build directories See issue https://github.com/nextstrain/.github/issues/63
| * | f9c3e198 fix: upper case root sequence for ancestral reconstruction
| |/  
* | 9839199f (origin/master, origin/HEAD) Remove outdated note "we have just updated augur to v6"
```

My suggested alternative:

```
*   9003db0b (HEAD -> master) Merge branch 'pr-1323-rebased'
|\  
| * 07fa7b5a (pr-1323-rebased) Add changelog entry
| * 5ff71be0 fix: upper case root sequence for ancestral reconstruction
| * 57d0123b Add failing test for #1323
* | 9839199f (origin/master, origin/HEAD) Remove outdated note "we have just updated augur to v6"
```

Given that this PR's scope is small, I think a squash merge is also fine:

```
* 12324ead (HEAD -> master) fix: upper case root sequence for ancestral reconstruction (#1323)
* 9839199f (origin/master, origin/HEAD) Remove outdated note "we have just updated augur to v6"
```

File: augur/ancestral.py
Comment (victorlin) [reply augur/ancestral.py:1]: I think you're right - at least for `augur index`, [sequences are lowercased](https://github.com/nextstrain/augur/blob/12324ead25434cdb92acc18e2f45f90cf0cf65d4/augur/index.py#L141).

File: CHANGES.md
Comment (corneliusroemer) [reply CHANGES.md:1]: Squash merge seems like the right thing to do here!

Comment (victorlin) [reply CHANGES.md:1]: Another note is that the current diff vs master (at least by the "Files changed" tab) contains d5e5dc32bf01163a7d067134aead2ff46e9bd454 which is a bit misleading. It gets dropped by squash merge since it's already in master as 5aba7ec9b3b1c3c676c3f2b850d1148061de7340, so that's not a problem.

PR seems good to me 👍

**Comment by corneliusroemer** [comment]
This was actually resolved by @jameshadfield in https://github.com/nextstrain/augur/commit/947e217382fa3655d30081d1e8bc532d6523e01a in an identical way so we can close this PR
